### PR TITLE
abstract base array for LArray

### PR DESCRIPTION
**Pull request by rafaqz**
If people think this is a good idea. 

This also cleans up a few errors I let back in debugging the last pull request...

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
I like the idea, just want to make sure it works.

**Comment by rafaqz** [comment]
Sure, me too.

File: src/larray.jl [hunk 1]
```diff
@@ -1,14 +1,14 @@
-struct LArray{T,N,Syms} <: DenseArray{T,N}
-  __x::Array{T,N}
-  LArray{Syms}(__x) where Syms = new{eltype(__x),ndims(__x),Syms}(__x)
-  LArray{T,N,Syms}(__x) where {T,N,Syms} = new{T,N,Syms}(__x)
+struct LArray{T,N,D<:AbstractArray{T,N},Syms} <: DenseArray{T,N}
```
Comment (devmotion) [inline src/larray.jl]: Wouldn't it make sense to be consistent and either use `D<:DenseArray{T,N}` or define `LArray{T,N,D<:AbstractArray{T,N},Syms} <: AbstractArray{T,N}`?

Comment (rafaqz) [reply src/larray.jl]: Yeah that should be consistent, and I don't think it makse sense to have a labelled sparse array. But does the limitation to `DenseArray` actually have any benefit?

**Comment by rafaqz** [comment]
Just realised I didn't update display.jl, and `show()` is breaking. I'll fix that up.

**Comment by rafaqz** [comment]
There are also some issues with similar. We need a few tests to check that the labels stay the same.

**Pull request merged**